### PR TITLE
Handle assert_never() when imported from typing_extensions

### LINKED ISSUE #9780: `possibly-used-before-assignment` does not treat `assert_never` the same when imported from `typing_extensions`
### Bug description

```python
The following code succeeds in the following environment using python 3.12:


pylint 3.2.5
astroid 3.2.2
Python 3.12.4 (main, Jun  7 2024, 06:33:07) [GCC 14.1.1 20240522]
```

but fails with python 3.10

```
pylint 3.2.5
astroid 3.2.2
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
```

```py
from enum import Enum
from typing import Literal
from typing_extensions import assert_never


class As(Enum):
    A = "A"
    AA = "AA"


a: Literal[As.A, As.AA] = As.A

if a == As.A:
    NUM_AS = 1
elif a == As.AA:
    NUM_AS = 2
else:
    assert_never(a)

print(NUM_AS)
```

### Configuration

_No response_

### Command used

```shell
./venv/bin/pylint --disable=all --enable=possibly-used-before-assignment test.py
```

### Pylint output

```python
either nothing (success) or

test.py:20:6: E0606: Possibly using variable 'NUM_AS' before assignment (possibly-used-before-assignment)
```

### Expected behavior

I expect pylint to detect that everything is right and it should not detect any issues.

Also, I expect pylint to behave consistent accross different python version

### Pylint version

```shell
pylint 3.2.5
```

### OS / Environment

Arch (with python 3.12)
Ubuntu (with python 3.10)

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description
Closes #9780

### CODE REVIEW

**Review by DanielNoord — DISMISSED** [review]

File: tests/functional/u/used/used_before_assignment_py311.py [hunk 1]
```diff
@@ -1,6 +1,12 @@
 """assert_never() introduced in 3.11"""
 from enum import Enum
-from typing import assert_never
+
+from pylint.constants import PY311_PLUS
+
+if PY311_PLUS:
```
Comment (Pierre-Sassoulas) [inline tests/functional/u/used/used_before_assignment_py311.py:6]: I think we should use sys.version_info directly here. Importing from pylint create coupling so more maintenance work when we drop python 3.11 and more importantely code in the wild will never import out internal constants so it makes functional tests less likely to be representative of real code.

**Review by Pierre-Sassoulas — COMMENTED** [review]
Élégant fix, I like when we remove complexity and fix bugs at the same time it's surprisingly often happening too. Just a nit.

File: tests/functional/u/used/used_before_assignment_py311.py (hunk 1 above)
Comment (jacobtylerwalls) [reply tests/functional/u/used/used_before_assignment_py311.py:6]: Why is it Less maintenance?

Comment (Pierre-Sassoulas) [reply tests/functional/u/used/used_before_assignment_py311.py:6]: We'll have to modify this when we drop python 3.11 and the constant do not exists anymore, right ?

Comment (jacobtylerwalls) [reply tests/functional/u/used/used_before_assignment_py311.py:6]: Yes, but I was suggesting it was the same whether we use the constant or the version guard.

Comment (Pierre-Sassoulas) [reply tests/functional/u/used/used_before_assignment_py311.py:6]: This is a functional test, I don't think functional test should look like internal pylint code. However it's not really important, let's move on

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**